lib/ansible/modules/cloud/vmware/vmware_evc_mode.py

```diff
@@ -0,0 +1,232 @@
+    default: present
+    type: str
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = '''
+    - name: Enable EVC Mode
+      vmware_evc_mode:
+         hostname: "{{ groups['vcsa'][0] }}"
+         username: "{{ vcenter_username }}"
+         password: "{{ site_password }}"
+         datacenter_name: "{{ datacenter_name }}"
+         cluster_name: "{{ cluster_name }}"
+         evc_mode: "intel-broadwell"
+         state: present
+      delegate_to: localhost
+      register: enable_evc
+
+    - name: Disable EVC Mode
+      vmware_evc_mode:
+         hostname: "{{ groups['vcsa'][0] }}"
+         username: "{{ vcenter_username }}"
+         password: "{{ site_password }}"
+         datacenter_name: "{{ datacenter_name }}"
+         cluster_name: "{{ cluster_name }}"
+         state: absent
+      delegate_to: localhost
+      register: disable_evc
+'''
+
+RETURN = """
+result:
+    description: information about performed operation
+    returned: always
+    type: str
+    sample: "EVC Mode for 'intel-broadwell' has been enabled."
+"""
+
+try:
+    from pyVmomi import vim
+except ImportError:
+    pass
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+from ansible.module_utils.vmware import (PyVmomi, find_datacenter_by_name, find_cluster_by_name,
+                                         vmware_argument_spec, wait_for_task, TaskError)
+
+
+class VMwareEVC(PyVmomi):
+    def __init__(self, module):
+        super(VMwareEVC, self).__init__(module)
+        self.cluster_name = module.params['cluster_name']
+        self.evc_mode = module.params['evc_mode']
+        self.datacenter_name = module.params['datacenter_name']
+        self.desired_state = module.params['state']
+        self.datacenter = None
+        self.cluster = None
+
+    def process_state(self):
+        """
+        Manage internal states of evc
+        """
+        evc_states = {
+            'absent': {
+                'present': self.state_disable_evc,
+                'absent': self.state_exit_unchanged,
+            },
+            'present': {
+                'present': self.state_update_evc,
+                'absent': self.state_enable_evc,
+            }
+        }
+        current_state = self.check_evc_configuration()
+        # Based on the desired_state and the current_state call
+        # the appropriate method from the dictionary
+        evc_states[self.desired_state][current_state]()
+
+    def check_evc_configuration(self):
+        """
+        Check evc configuration
+        Returns: 'Present' if evc enabled, else 'absent'
+        """
+        try:
+            self.datacenter = find_datacenter_by_name(self.content, self.datacenter_name)
+            if self.datacenter is None:
+                self.module.fail_json(msg="Datacenter '%s' does not exist." % self.datacenter_name)
+            self.cluster = self.find_cluster_by_name(cluster_name=self.cluster_name)
```

Comment:
Pass self.datacenter to cluster find API - 

```suggestion
            self.cluster = self.find_cluster_by_name(cluster_name=self.cluster_name, datacenter_name=self.datacenter)
```